src/ayab/encoders.cpp

```diff
@@ -155,6 +150,38 @@ void Encoders::encA_rising() {
       (hallValue > FILTER_L_MAX[static_cast<uint8_t>(m_machineType)])) {
     m_hallActive = Direction_t::Left;
 
+    // Only set the belt shift the first time a magnet passes the turn mark.
+    // Headed to the right.
+    if (!m_passedLeft && Direction_t::Right == m_direction) {
+      // Belt shift signal only decided in front of hall sensor
+      m_beltShift = digitalRead(ENC_PIN_C) != 0 ? BeltShift::Shifted : BeltShift::Regular;
+      m_passedLeft = true;
+
+      if (Carriage_t::Garter == m_carriage) {
+        // This has to be the first magnet and the belt shift needs to be swapped
+        // But only for the G-carraige
```

Comment:
oops, *carraige*, looks like it's the last one.
```suggestion
        // But only for the G-carriage
```